crypto/ocsp/ocsp_test.cc

```diff
@@ -218,3 +525,101 @@ TEST(OCSPTest, TestBasic) {
   basic_response = bssl::UniquePtr<OCSP_BASICRESP>(OCSP_response_get1_basic(ocsp_response.get()));
   ASSERT_TRUE(basic_response);
 }
+
+TEST(OCSPTest, TestGoodOCSP) {
+  bssl::UniquePtr<OCSP_RESPONSE> ocsp_response;
+  bssl::UniquePtr<OCSP_BASICRESP> basic_response;
+
+  ocsp_response = LoadOCSP_RESPONSE(ocsp_response_der);
+  ASSERT_TRUE(ocsp_response);
+
+  int ocsp_status = OCSP_response_status(ocsp_response.get());
+  ASSERT_EQ(OCSP_RESPONSE_STATUS_SUCCESSFUL, ocsp_status);
+
+  basic_response = bssl::UniquePtr<OCSP_BASICRESP>(OCSP_response_get1_basic(ocsp_response.get()));
+  ASSERT_TRUE(basic_response);
+
+  // Set up trust store and certificate chain
+  bssl::UniquePtr<X509_STORE> trust_store(X509_STORE_new());
+  X509_STORE_add_cert(trust_store.get(),LoadX509fromPEM(ca_cert).get());
+  bssl::UniquePtr<STACK_OF(X509)> server_cert_chain = CertsToStack(
+      {LoadX509fromPEM(server_cert).get(),LoadX509fromPEM(ca_cert).get()});;
+
+  X509 *subject = sk_X509_value(server_cert_chain.get(), 0);
+  X509 *issuer = nullptr;
+  // Find the issuer in the certificate chain. If it's not there, fail everything.
```

Comment:
Shouldn't we know this? Why do we have to look for it?

Reply:
I think this is just an extra check to find the issuer in the cert chain, as s2n is doing here: https://github.com/aws/s2n-tls/blob/main/tls/s2n_x509_validator.c#L444-L456 
s2n also has unit tests that test against an invalid issuer, so I assumed that this is something we shouldn't always know by default. https://github.com/aws/s2n-tls/blob/main/tests/unit/s2n_x509_validator_test.c#L1078-L1124 
We could argue that this is outside the scope of OCSP and more of a certificate validation phase that s2n does on its own? If so, hard setting the issuer would make sense.

Reply:
They have to support arbitrary chains. For this unit test, we don't.  Our chain is very simple:
```
  bssl::UniquePtr&lt;STACK_OF(X509)&gt; server_cert_chain = CertsToStack(      {LoadX509fromPEM(server_cert).get(),LoadX509fromPEM(ca_cert).get()});;
```
So, we can just skip this and index directly for what we need.